### ISSUE TITLE
Duplicate classes using mapbox 11.0.0-beta.1 and mapbox for android auto

### ISSUE THREAD

**Issue by FraRo96** [mapbox/mapbox-maps-android#2195]
## Environment

- Maps SDK Version: com.mapbox.navigation:ui-androidauto:0.22.0, com.mapbox.maps:android:11.0.0-beta.1, com.mapbox.extension:maps-compose:0.1.0

## Observed behavior and steps to reproduce

I use mapbox extension-compose (com.mapbox.maps:android:11.0.0-beta.1 + com.mapbox.extension:maps-compose:0.1.0). When I add mapbox for Android Auto dependencies (com.mapbox.navigation:ui-androidauto:0.22.0), I got this duplicate classes:

Duplicate class com.mapbox.common.module.okhttp.CallbackWrapper found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.CallbackWrapper$RequestCallback found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.CertificatePinnerFactory found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.HttpCallback found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.IdGenerator found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.LazyClient found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.MapboxOkHttpService found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.MapboxOkHttpService$1 found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.MapboxOkHttpService$HttpServiceOfflineSwitchObserver found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.NetworkUsageListener found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.NetworkUsageListener$1 found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.NetworkUsageListener$DummyEventListener found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.common.module.okhttp.NetworkUsageListener$NetworkUsageMetricCallback found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)
Duplicate class com.mapbox.module.Mapbox_HttpClientModuleConfiguration found in modules common-24.0.0-beta.2-runtime (com.mapbox.common:common:24.0.0-beta.2) and okhttp-23.3.1-runtime (com.mapbox.common:okhttp:23.3.1)

## Expected behavior
We should be able to use both compose and android auto extension in the same app.

Found a similar issue [here](https://github.com/mapbox/mapbox-maps-android/issues/2181) regarding use of the 11.0.0-beta.1.
Please, consider to fix as we're at an advanced development stage and we'd like to use Mapbox for Android Auto.

**Comment by dudeuter** [comment]
@FraRo96 can you share the gradle configuration for all of the Mapbox SDKs that you are using? It's likely that you simply have a conflicting version of our shared common SDK.

**Comment by FraRo96** [comment]
@dudeuter the gradle configuration I'm using is exactly the one I've reported above.
The conflict is in 11.0.0-beta.1 and ui-androidauto:0.22.0.
You can easily check by doing below steps:
1) create a new blank Android Project
2) add into app's build.gradle the following:
`    implementation("com.mapbox.navigation:ui-androidauto:0.22.0")
    implementation ("com.mapbox.maps:android:11.0.0-beta.1")`
3) get a bunch of "duplicate classes" build fails.

**Comment by FraRo96** [comment]
@dudeuter any feedback or news? Thank you.

**Comment by pengdev** [comment]
Hey @FraRo96 the compose extension is only compatible with v11 Maps SDK at the moment, we don't provide backward compatibility with v10 Maps SDK, or with Navigation SDK v2.x.

Future major version of navigation SDK will provide compatibility with Maps SDK v11 and till then you can use compose extension in your application.

Question: Do you use full navigation SDK or just `ui-androidauto` module?

**Comment by pengdev** [comment]
> Currently I'm just using ui-androidauto. In the next future I would go for full navigation SDK.

@FraRo96  glad to hear :) 

> I understand compose is just working with v11 beta. Still I don't get why it should not be compatible with Android Auto which is completely independent from. I literally use them in different independent modules: Compose for smartphone-app module, Android Auto for android-auto-app module.

Both Android Auto extension and Compose extension are built as extensions to the Maps SDK, and they rely on Maps SDK APIs. Even though we tried to minimise the API change between v10 and v11, there are still some breaking API changes making them not backwards compatible. The class duplication you observed is actually coming from our underlying transitive dependencies, they follow the same semver, meaning as this is a major release, we have introduced other breaking APIs across our stack. It would be impossible/really painful to solve the compatibility issue between major SDK versions. I think you'd have to wait for the release for a compatible version of Android Auto(Nav) SDK.
